### PR TITLE
chore: update issue templates

### PR DESCRIPTION
**Pull request by yizha1**
Updates:
- fix typo
- remove notary v2 reference
- remove unnecessary options

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

**Review by FeynmanZhou — DISMISSED** [review]
LGTM

**Review by patrickzheng200 — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/bug-or-issue.yaml [hunk 1]
```diff
@@ -27,8 +27,6 @@ body:
       label: "What is the areas you experience the issue in?"
       options: 
         - Notation CLI
-        - Notation-Core-Go Library
```
Comment (priteshbandi) [inline .github/ISSUE_TEMPLATE/bug-or-issue.yaml:30]: If there is only one option should we just remove this question?

File: .github/ISSUE_TEMPLATE/feature-request.yaml [hunk 2]
```diff
@@ -27,8 +27,6 @@ body:
       label: "What is the areas you would like to add the new feature to?"
       options: 
         - Notation CLI
```
Comment (priteshbandi) [inline .github/ISSUE_TEMPLATE/feature-request.yaml:29]: ditto

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**